src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -160,19 +159,12 @@ public bool TryGetQuantityCodeId(string code, out int quantityCodeId)
             return _quantityCodeToId.TryGetValue(code, out quantityCodeId);
         }
 
-        public async Task StartAsync(CancellationToken cancellationToken)
+        public async Task EnsureInitialized()
         {
             ThrowIfCurrentSchemaVersionIsNull();
 
             // If the fhir-server is just starting up, synchronize the fhir-server dictionaries with the SQL database
             await Initialize((int)_schemaInformation.Current, true, CancellationToken.None);
-
-            await _mediator.Publish(new StorageInitializedNotification(), cancellationToken);
```

Comment:
Other services now depend on this notification

Reply:
This is the other pr for reference #1831

Reply:
Thanks! Yeah moved it in the initialization method. And it works now as its not triggered until database is not setup with schema version >=1